core/arch/arm/mm/core_mmu.c

```diff
@@ -97,9 +97,13 @@ register_phys_mem(MEM_AREA_TEE_RAM_RW, VCORE_UNPG_RW_PA, VCORE_UNPG_RW_SZ);
 #ifdef CFG_WITH_PAGER
 register_phys_mem(MEM_AREA_TEE_RAM_RX, VCORE_INIT_RX_PA, VCORE_INIT_RX_SZ);
 register_phys_mem(MEM_AREA_TEE_RAM_RO, VCORE_INIT_RO_PA, VCORE_INIT_RO_SZ);
-#endif
-#else
+#endif /*CFG_WITH_PAGER*/
+#else /*!CFG_CORE_RWDATA_NOEXEC*/
 register_phys_mem(MEM_AREA_TEE_RAM, CFG_TEE_RAM_START, CFG_TEE_RAM_PH_SIZE);
+#endif /*!CFG_CORE_RWDATA_NOEXEC*/
+
+#if defined(CFG_CORE_SANITIZE_KADDRESS) && defined(CFG_WITH_PAGER)
+register_phys_mem(MEM_AREA_TEE_ASAN, ASAN_MAP_PA, ASAN_MAP_SZ);
```

Comment:
A comment to state `/* Asan ram is part of MEM_AREA_TEE_RAM_RW when pager is disable */` ?